packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts

```diff
@@ -63,32 +71,54 @@ function create(context: Rule.RuleContext): Rule.RuleListener {
     }
   });
 
-  const cwd = normalizeCwd(context.getCwd ? context.getCwd() : undefined);
-  const turboConfig = options?.[0]?.turboConfig;
-  const turboVars = getEnvVarDependencies({
-    cwd,
-    turboConfig,
-  });
+  const cwd = normalizeCwd(
+    context.getCwd ? context.getCwd() : undefined,
+    options
+  );
+  const filePath = getPhysicalFilename();
+  const allTurboVars =
+    getEnvVarDependencies({
+      cwd,
+    }) || {};
+
+  const globalTurboVars = allTurboVars["//"];
+
+  // find any workspace configs that match the current file path
+  // find workspace config (if any) that match the current file path
+  const workspaceKey = Object.keys(allTurboVars).find(
+    (workspacePath) => filePath !== "//" && filePath.startsWith(workspacePath)
+  );
+
+  let workspaceTurboVars: Set<string> | null = null;
+  if (workspaceKey) {
+    workspaceTurboVars = allTurboVars[workspaceKey];
+  }
 
   // if this returns null, something went wrong reading from the turbo config
   // (this is different from finding a config with no env vars present, which would
   // return an empty set) - so there is no point continuing if we have nothing to check against
-  if (!turboVars) {
+  if (!globalTurboVars) {
     // return of {} bails early from a rule check
     return {};
   }
 
   const checkKey = (node: Node, envKey?: string) => {
     if (
       envKey &&
-      !turboVars.has(envKey) &&
+      !globalTurboVars.has(envKey) &&
       !regexAllowList.some((regex) => regex.test(envKey))
     ) {
-      context.report({
-        node,
-        message: "${{ envKey }} is not listed as a dependency in turbo.json",
-        data: { envKey },
-      });
+      // if we have a workspace config, check that too
+      if (workspaceTurboVars && workspaceTurboVars.has(envKey)) {
+        return {};
+      } else {
+        context.report({
+          node,
+          message:
+            "${{ envKey }} is not listed as a dependency in any turbo.json",
```

Comment:
not a blocker, but it would probably be helpful to say which workspaces we looked in